### PR TITLE
feat: rewrite functional spreadsheet project

### PR DESCRIPTION
**Pull request by naomi-lgbt**
Checklist:

- [ ] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [ ] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [ ] My pull request targets the `main` branch of freeCodeCamp.
- [ ] I have tested these changes either locally on my machine, or GitPod.

Closes #XXXXX

### CODE REVIEW

**Comment by bbsmooth** [comment]
Hey, I know accessibility is not the main focus of this course so I'm not going to suggest we try to implement a truly accessible grid. The only thing I will suggest is that we should reinforce that every `input` needs an accessible name. In this case, I think we should use `aria-label` and set it to the same value as the `id`.

**Comment by naomi-lgbt** [comment]
As much as I know accessibility is important, I do worry about the complexity of this project and piling additional information on it.

How much would it take for a properly accessible grid?

**Comment by bbsmooth** [comment]
> I do worry about the complexity of this project

Completely agree. I am not insisting we implement an accessible grid. Just giving the `input`s an `aria-label` will be good enough.

If we did want to implement a true grid it would require at a minimum preventing the tab key from working inside the grid and instead handling the arrow keys to navigate amongst the cells (which would include the concept of a roving tabindex). You can see some examples at https://www.w3.org/WAI/ARIA/apg/patterns/grid/.

**Review by jdwilkin4 — CHANGES_REQUESTED** [review]
I have finished going through steps 1-50. Will finish going through the remaining steps this week. 
It looks good 👍 
Left a couple of comments

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-functional-programming-by-building-a-spreadsheet/643498755d54c6279ba09078.md [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+Your `sum` function should use arrow syntax.
+
+```js
+assert.match(code, /const\s*sum\s*=\(?.*\)?\s*=>/);
+```
+
+Your `sum` function should have a `nums` parameter.
+
+```js
+assert.match(code, /const\s*sum\s*=\(?\s*nums\s*\)?\s*=>/);
```
[the remark this answers is not in the document]
Comment (naomi-lgbt) [reply curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-functional-programming-by-building-a-spreadsheet/643498755d54c6279ba09078.md:43]:
```suggestion
assert.match(code, /const\s*sum\s*=\s*\(?\s*nums\s*\)?\s*=>/);
```

Just a forgotten space - this should fix it.

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-functional-programming-by-building-a-spreadsheet/646d4a07a8fb14d55cd70e09.md [hunk 2]
```diff
@@ -0,0 +1,198 @@
+---
+id: 646d4a07a8fb14d55cd70e09
+title: Step 97
+challengeType: 0
+dashedName: step-97
+---
+
+# --description--
+
+Add `someeven` and `everyeven` properties, which check if `some` or `every` element in the `nums` array is even, respectively.
```
Comment (jdwilkin4) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-functional-programming-by-building-a-spreadsheet/646d4a07a8fb14d55cd70e09.md:10]: I think this is the first time where the `arr.some()` and `arr.every()` methods are introduced. 
If that is the case, then a brief explanation and example will be needed here

**Review by jdwilkin4 — CHANGES_REQUESTED** [review]
I was able finish going through the rest of the steps. 
I think it looks really good.
Left two small comments 👍

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-functional-programming-by-building-a-spreadsheet/646d40fe4b7b50c30c2b4cd8.md [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+title: Step 75
+challengeType: 0
+dashedName: step-75
+---
+
+# --description--
+
+Your `infixEval` function will only evaluate the first multiplication or division operation, because `regex` isn't global. This means you'll want to use a recursive approach to evaluate the entire string.
+
+If `infixEval` does not find any matches, it will return the `str` value as-is. Using a ternary expression, check if `str2` is equal to `str`. If it is, return `str`, otherwise return the result of calling `highPrecedence()` on `str2`.
```
[the remark this answers is not in the document]
Comment (naomi-lgbt) [reply curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-functional-programming-by-building-a-spreadsheet/646d40fe4b7b50c30c2b4cd8.md:12]: Good catch - I was erroneously looking for `str2 === str1`.

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-functional-programming-by-building-a-spreadsheet/646d4a07a8fb14d55cd70e09.md (hunk 2 above)
Comment (naomi-lgbt) [reply curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-functional-programming-by-building-a-spreadsheet/646d4a07a8fb14d55cd70e09.md]: Split into two steps and expanded the explanations.

**Review by jdwilkin4 — APPROVED** [review]
LGTM! 
Happy to approve 👍

**Review by raisedadead — APPROVED** [review]

**Pull request merged**